src/SemiFungible1155.sol

```diff
@@ -222,29 +280,19 @@ contract SemiFungible1155 is Upgradeable1155 {
     ) public override {
         if (_to == address(0x0)) revert ToZeroAddress();
         if (_from != msg.sender) revert NotApprovedOrOwner(); //TODO Allowance approval
+        if (getNonFungibleIndex(_id) == 0) revert NotAllowed();
 
         uint256 tokenValue = tokenValues[_id];
+        uint256 typeID = getNonFungibleBaseType(_id);
 
-        //TODO emit event per case? Since value NF should be 1
-        //TODO Block marketplace transfer of claim data ownership
-        if (getNonFungibleIndex(_id) == 0) {
-            revert NotAllowed();
-        } else {
-            uint256 typeID = getNonFungibleBaseType(_id);
-
-            tokenUserBalances[typeID][_from] -= tokenValue;
-            tokenUserBalances[typeID][_to] += tokenValue;
-
-            tokenUserBalances[_id][_from] -= tokenValue;
-            tokenUserBalances[_id][_to] += tokenValue;
-        }
+        //TODO evaluate need of double accounting (base type & fractions)
+        tokenUserBalances[typeID][_from] -= tokenValue;
```

Comment:
Do we need to update `owners`?

Reply:
Yes, done. Adding the checks to the tests made them to long for comfort so also added a few test helpers. So lot's of changes, but primarily in the test suite.